api/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+enum RunnerType {
+  DOCKER = 0;
+}
+
+message Action {
+  string action_id = 1;
+  RunnerType runner_type = 2;
+  optional string image = 3;
+  repeated string commands = 4;
+}
+
+message RegisteredAgent {
+  string hostname = 1;
+  int32 cpu = 2;
+  int32 memory = 3;
+}
+
+message AgentId {
+  string id = 1;
+}
+
+message ActionLog {
+  string action_id = 1;
+  string log = 2;
+  optional bool success = 3; // Only populated if action is complete
+  optional int32 error_code = 4; // Only populated if success is false
+}
+
+message HealthStatus {
```

Comment:
As a member of agent team, for HealthStatus we should implement the same interface. Not just copy/paste. I don't know if it is possible but it would make things easier.
Maybe add the number of currently running jobs ?

Reply:
The agent status is going to be a scheduler service, so the corresponding message should be defined here.

Reply:
I don't think the number of jobs is relevant since it does not provide any information about the workload of the agent